### PR TITLE
Feat/split oracletx

### PR DESCRIPTION
**Pull request by sh-cha**
## Release Notes

- **New Features**
  - Enhanced configuration structure for Challenger and Executor components, improving clarity and organization.
  - Introduced sender information in batch message handling and processing across various components.
  - Added new querying capabilities for grants and executors in the BaseChild struct.
  - Implemented new methods for retrieving account addresses and managing oracle accounts in the BaseChild and BaseHost structs.
  - Added a new command for transaction operations, including granting oracle permissions.

- **Bug Fixes**
  - Improved error handling in multiple methods, ensuring more informative error messages.

- **Refactor**
  - Restructured message queues in both BaseChild and BaseHost to allow for sender-specific message handling.
  - Updated method signatures across various components to include additional return values for improved context.
  - Simplified address retrieval methods and enhanced transaction handling logic in the Broadcaster struct.
  - Removed outdated address retrieval methods from BaseChild and BaseHost for cleaner code.
  - Streamlined the QueryWithdrawal method for better error handling and response structure.

- **Chores**
  - Updated dependency versions in `go.mod` for better stability and performance.

### CODE REVIEW

**Comment by beer-1** [comment]
need to set Sender when you append processed message

like

```go
executor/host/handler.go#62

func (h *Host) txHandler(_ context.Context, args nodetypes.TxHandlerArgs) error {
	if args.BlockHeight == args.LatestHeight && args.TxIndex == 0 {
		if msg, err := h.oracleTxHandler(args.BlockHeight, args.Tx); err != nil {
			return err
		} else if msg != nil {
			h.AppendProcessedMsgs(btypes.ProcessedMsgs{
				Msgs:      []sdk.Msg{msg},
				Timestamp: time.Now().UnixNano(),
				Save:      false,
			})
		}
	}
	return nil
}

```

File: provider/host/host.go [hunk 1]
```diff
@@ -34,7 +35,8 @@ type BaseHost struct {
 	ophostQueryClient ophosttypes.QueryClient
 
 	processedMsgs []btypes.ProcessedMsgs
-	msgQueue      []sdk.Msg
+	msgQueue      map[string][]sdk.Msg
```
Comment (beer-1) [inline provider/host/host.go:38]: is there any risk to access this concurrently?

File: provider/host/host.go [hunk 2]
```diff
@@ -154,16 +156,21 @@ func (b BaseHost) DB() types.DB {
...
 
-func (b BaseHost) GetMsgQueue() []sdk.Msg {
+func (b BaseHost) GetMsgQueue() map[string][]sdk.Msg {
 	return b.msgQueue
 }
 
-func (b *BaseHost) AppendMsgQueue(msg sdk.Msg) {
-	b.msgQueue = append(b.msgQueue, msg)
+func (b *BaseHost) AppendMsgQueue(msg sdk.Msg, sender string) {
+	if b.msgQueue[sender] == nil {
```
Comment (beer-1) [inline provider/host/host.go]: we can remove this if statement, `append(arr(nil), item)` should works

File: provider/host/msgs.go [hunk 3]
```diff
@@ -53,7 +53,7 @@ func (b BaseHost) CreateBatchMsg(batchBytes []byte) (sdk.Msg, error) {
 	)
 	err = msg.Validate(b.node.AccountCodec())
 	if err != nil {
-		return nil, err
+		return nil, "", err
 	}
-	return msg, nil
+	return msg, "", nil
```
Comment (beer-1) [inline provider/host/msgs.go]: don't we need to return sender?

File: provider/host/msgs.go [hunk 4]
```diff
@@ -32,18 +32,18 @@ func (b BaseHost) GetMsgProposeOutput(
...
```
Comment (beer-1) [inline provider/host/msgs.go]: here too

File: provider/host/host.go (hunk 1 above)
Comment (sh-cha) [reply provider/host/host.go:38]: no, it's used only in single thread

**Review by beer-1 — APPROVED** [review]
LGTM with minor comments

**Pull request merged**